buildSrc/src/main/java/org/opensearch/gradle/pluginzip/Publish.java

```diff
@@ -63,12 +64,19 @@ static String getProperty(String name, Project project) {
     @Override
     public void apply(Project project) {
         this.project = project;
-        project.afterEvaluate(evaluatedProject -> { configMaven(project); });
-        project.getGradle().getTaskGraph().whenReady(graph -> {
-            if (graph.hasTask(LOCALMAVEN)) {
-                project.getTasks().getByName(PLUGIN_ZIP_PUBLISH_POM_TASK).setEnabled(false);
+        project.afterEvaluate(evaluatedProject -> {
+            configMaven(project);
+            Task validatePluginZipPom = project.getTasks().findByName("validatePluginZipPom");
+            if (validatePluginZipPom != null) {
+                project.getTasks().getByName("validatePluginZipPom").mustRunAfter("generatePomFileForNebulaPublication");
```

Comment:
Just curious, why  `mustRunAfter` but `dependsOn`? We do not rely on any order, right?

Reply:
`dependsOn` does the same, `mustRunAfter` mandates that task has to run **exactly in order** after the task in question: I don't think we have such exact order requirements, `dependsOn` should be sufficient

Reply:
`mustRunAfter` would ensure that task has been called out before those tasks. @reta

Reply:
I would suggest to switch to `dependsOn`, it introduces implicit dependencies (which is what we need, because our publication depends on it) [1]:

> A task may have dependencies on other tasks or might be scheduled to always run after another task. Gradle ensures that all task dependencies and ordering rules are honored when executing tasks, so that the task is executed after all of its dependencies and any "must run after" tasks have been executed.

but `mustRunAfter` does merely controls the order [2]:

> When you use the “must run after” ordering rule you specify that taskB must always run after taskA, whenever both taskA and taskB will be run. 


The difference is actually very easy to demonstrate: if you just run a single `publishPluginZipPublicationToZipStagingRepository` task, none of the `mustRunAfter` tasks  **will not be** executed (notice `generatePomFileForNebulaPublication` is not there):


```
./gradlew -m publishPluginZipPublicationToZipStagingRepository 
:generateNotice SKIPPED                                                                        
:compileJava SKIPPED                                                                                                                                                                          
:processResources SKIPPED                                                                      
:classes SKIPPED                                                                               
:jar SKIPPED                                                                                   
:copyPluginPropertiesTemplate SKIPPED                                                          
:pluginProperties SKIPPED                                                                      
:bundlePlugin SKIPPED                                                                          
:generatePomFileForPluginZipPublication SKIPPED                                                
:publishPluginZipPublicationToZipStagingRepository SKIPPED     
```

Now, using `dependsOn` (notice `generatePomFileForNebulaPublication` is there):

```
./gradlew -m publishPluginZipPublicationToZipStagingRepository 
:generateNotice SKIPPED
:compileJava SKIPPED
:processResources SKIPPED
:classes SKIPPED
:jar SKIPPED
:copyPluginPropertiesTemplate SKIPPED
:pluginProperties SKIPPED
:bundlePlugin SKIPPED
:generatePomFileForNebulaPublication SKIPPED
:generatePomFileForPluginZipPublication SKIPPED
:publishPluginZipPublicationToZipStagingRepository SKIPPED

```

Hope it is sufficient to justify `dependsOn` usage.

[1] https://docs.gradle.org/current/dsl/org.gradle.api.Task.html#N18A16
[2] https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:ordering_tasks

Reply:
> But both works here slightly_smiling_face

I hope the examples above help to clarify when it does not work

Reply:
Thanks for the info @reta .
`-m` shows the task actions disabled, here our initial error is because we have a task `generatePomFileForNebulaPublication` that has been not called and these `validatePluginZipPom` & `publishPluginZipPublicationToZipStagingRepository`  actually uses the output of `generatePomFileForNebulaPublication`,  using `mustRun` will ensure this task `generatePomFileForNebulaPublication` has been invoked when ever the tasks  `validatePluginZipPom` & `publishPluginZipPublicationToZipStagingRepository` are triggered, which should solve the output dependency, hence I mentioned running `build` should work, even though it does not show with `-m` output `mustRun` ensures it will execute the task. 

But that said `dependsOn` should also work as we are adding exclusive graph dependency, I have updated my latest commit with `dependsOn` and ran few plugins `build.sh` and runs fine, please check @reta